### PR TITLE
Add the OS type to the output and intermediate paths

### PR DESCRIPTION
**Pull request by mmitche**
This avoids issues where switching between Unix and Windows_NT causes rebuild problems

### CODE REVIEW

File: dir.props [hunk 1]
```diff
@@ -75,16 +75,17 @@
   <!-- Setup the default output and intermediate paths -->
   <PropertyGroup>
     <BaseOutputPath>$(ProjectDir)bin\</BaseOutputPath>
-    <BaseOutputPathWithConfig>$(BaseOutputPath)$(Configuration)\</BaseOutputPathWithConfig>
-    <BaseOutputPathWithConfig Condition="'$(Platform)' != 'AnyCPU'">$(BaseOutputPath)$(Platform)\$(Configuration)\</BaseOutputPathWithConfig>
+    <Error Condition="'$(OS)' == ''">Need OS</Error>
```
Comment (ellismg) [inline dir.props]: Could you make this a little more descriptive?  e.g "The 'OS' Property must be defined?

Comment (mmitche) [inline dir.props]: Sorry, that wasn't meant to make it in there...updated

**Comment by ellismg** [comment]
LGTM.

**Comment by AlexGhiondea** [comment]
LGTM

**Pull request merged**